src/ert/job_queue/queue.py

```diff
@@ -82,31 +82,24 @@ class JobQueue(BaseCClass):  # type: ignore
     _alloc = ResPrototype("void* job_queue_alloc()", bind=False)
     _free = ResPrototype("void job_queue_free( job_queue )")
     _set_driver = ResPrototype("void job_queue_set_driver( job_queue , void* )")
-    _num_running = ResPrototype("int  job_queue_get_num_running( job_queue )")
-    _num_complete = ResPrototype("int  job_queue_get_num_complete( job_queue )")
-    _num_waiting = ResPrototype("int  job_queue_get_num_waiting( job_queue )")
-    _num_pending = ResPrototype("int  job_queue_get_num_pending( job_queue )")
+
+    _num_jobs = ResPrototype("int job_queue_get_num_status( job_queue, int )")
 
     _add_job = ResPrototype("int job_queue_add_job_node(job_queue, job_queue_node)")
 
-    def __repr__(self) -> str:
-        nrun, ncom, nwait, npend = (
-            self.num_running,
-            self.num_complete,
-            self.num_waiting,
-            self.num_pending,
-        )
-        return self._create_repr(  # type: ignore
-            f"num_running={nrun}, num_complete={ncom}, "
-            f"num_waiting={nwait}, num_pending={npend}"
+    def _get_count_status(self) -> str:
+        return (
+            f"num_running={self._num_jobs(JobStatusType.JOB_QUEUE_RUNNING)}, "
+            f"num_complete={self._num_jobs(JobStatusType.JOB_QUEUE_SUCCESS)}, "
+            f"num_waiting={self._num_jobs(JobStatusType.JOB_QUEUE_WAITING)}, "
+            f"num_pending={self._num_jobs(JobStatusType.JOB_QUEUE_PENDING)}"
         )
 
+    def __repr__(self) -> str:
+        return self._create_repr(self._get_count_status())  # type: ignore
```

Comment:
Could we have a quick little test of `__repr__` and `__str__`:

def test_job_queue_str():
    local_driver = ...
    assert repr(JobQueue(local_driver)) == "..."
    assert str(JobQueue(local_driver)) == "..."